craft_application/models/constraints.py

```diff
@@ -36,6 +36,13 @@ class ProjectName(ConstrainedStr):
     regex = re.compile(r"^([a-z0-9][a-z0-9-]?)?[a-z]+([a-z0-9-]?[a-z0-9])*$")
 
 
+MESSAGE_INVALID_NAME = (
```

Comment:
I thought it would be handy to have the error message right next to the regex, so we can compare the two